lib/chef/knife/yaml_convert.rb

```diff
@@ -0,0 +1,66 @@
+#
+# Author:: Bryan McLellan <btm@loftninjas.org>
+# Copyright:: Copyright 2020, Chef Software Inc.
+# License:: Apache License, Version 2.0
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+require_relative "../knife"
+class Chef::Knife::YamlConvert < Chef::Knife
+
+  banner "knife yaml convert YAML_FILENAME"
+
+  def run
+    if name_args.empty?
+      ui.error("Please specify the file name of a YAML recipe to convert to Ruby")
+      exit 1
+    elsif name_args.size >= 2
+      ui.error("Only one recipe may converted at a time")
+      exit 1
+    end
+
+    filename = @name_args[0]
+    yaml_contents = IO.read(filename)
+
+    if ::YAML.load_stream(yaml_contents).length > 1
```

Comment:
Shouldn't this require yaml since it uses it?

Reply:
I did at first, and then I noticed that @lamont-granquist didn't add it in #8653, so thought maybe it was one of those newer things about how we're doing requires that I didn't understand, but this morning I realize it was probably just an oversight.